lib/signatures.ts

```diff
@@ -1,25 +1,34 @@
 import { promises } from 'fs';
-import { SignatureHashAlgorithm, SignatureResult } from './types';
+import { SignatureFormat, SignatureResult } from './types';
 import { getDubHashSignature } from './dubhash';
 import { getUhashSignature } from './uhash';
 import pMap = require('p-map');
 import { join } from 'path';
 
 const { readFile } = promises;
 
+type Signer = (path: string, content: Buffer) => SignatureResult | null;
+interface SignatureFormatMap {
+  [SignatureFormat: string]: Signer;
+}
+
+const algorithms: SignatureFormatMap = {
```

Comment:
I appreciate the introduction of a map abstraction here, even so I believe that is too early for us to introduce this abstraction as our goal for the moment is to replace `dubhash` with `uhash`, if later on we will decide of have multiple algorithms then we can abstract. i.e. KISS principle here